### DISCUSSION TITLE
Blade Component in Header

### DISCUSSION THREAD

**Discussion by abhaypithadiya** [Power-Components/livewire-powergrid#619] [Q&A]
Hey! Can we add an bladecomponent in the header. For example. Bulk-delete. Can I use a customize button?

**Comment by rogermedico** [comment]
Hello,

I need this feature also and I have been investigating through the code. 

What I found is that you can add a Button::add('whatever')->bladecomponent() in header() method. But then in file https://github.com/Power-Components/livewire-powergrid/blob/4.x/resources/views/components/actions-header.blade.php there is no @elseif($action->bladeComponent !== '').

I have published those files and added myself the @elseif directive and then inside that elseif you have to put a dynamic blade component like this:

`<x-dynamic-component
    :component="$action->bladeComponent"
    :attributes="$parameters"
/>`

/roger

**Comment by kolydart** [comment]
A question on this, before I try to implement it. Can blade components in header be used to create dynamic filters? A type of dynamic filters could be faceted search.

**Comment by rogermedico** [comment]
Hello,

I have made two pull requests and this feature is now part of official package (update package to 4.8.4). I implemented blade component with some params in header in my app successfully. So, I don't know what you mean by "dynamic filter" but you can try it.

Let us know!

**Reply by kolydart** [reply]
Faceted search is a technique used in platforms like eBay, where users can apply multiple filters (such as price, brand, condition, and location) simultaneously to narrow down search results and find specific items more efficiently. Every time you select a filter, the other filters adapt to remove values that do not exist in current found set. Something like [this](http://demo-eshop-filters.livewirekit.com/bootstrap), but with select inputs & removing values with zero occurencies in current set.
Was this better? Do you think this can be implemented with your pull requests? Or, do you have a working demo example on your app so I can figure out myself?

**Comment by abhaypithadiya** [comment]
Hey @rogermedico, can you check this issue https://github.com/Power-Components/livewire-powergrid/issues/1083? Can you help me out here?
